### PR TITLE
Bundle show --outdated shouldn't be updating versions 

### PR DESCRIPTION
**Pull request by colby-swandale**
This PR is for #5375. I have added a spec (currently failing) of the desired behavior for `bundle show --outdated`

I'm currently looking into a fix for this but welcome anyone else to come in as well.

### CODE REVIEW

File: spec/commands/show_spec.rb [hunk 1]
```diff
@@ -161,4 +161,31 @@
...
+      build_repo2
+    end
+
+    it "doesn't update gems to newer versions" do
+      install_gemfile <<-G
+        source "file://#{gem_repo2}"
+        gem "rails"
+      G
+
+      bundle "install"
```
Comment (segiddins) [inline spec/commands/show_spec.rb]: Redundant, previous method call will install for us

**Comment by colby-swandale** [comment]
@segiddins Seeing this bug has been hidden for quite a long time and is intended to be deprecated, should we just `resolve_with_cache!` instead of `resolve_remotely!`? Obviously it won't tell the user if the gem is actually up to date without pulling from rubygems but i think its more suitable than making any larger change.

**Comment by segiddins** [comment]
I highly doubt that will fix things. I can look into this next month, my guess is we just need to track down what's calling `Definition#lock`

**Comment by segiddins** [comment]
```diff
diff --git a/lib/bundler/cli/show.rb b/lib/bundler/cli/show.rb
index 77e845a60..47d4470ae 100644
--- a/lib/bundler/cli/show.rb
+++ b/lib/bundler/cli/show.rb
@@ -64,6 +64,7 @@ module Bundler
       else
         definition.resolve_with_cache!
       end
+      Bundler.reset!
       definition.specs
     end
 
diff --git a/spec/commands/show_spec.rb b/spec/commands/show_spec.rb
index 901d8acb7..fb9731909 100644
--- a/spec/commands/show_spec.rb
+++ b/spec/commands/show_spec.rb
@@ -169,21 +169,23 @@ RSpec.describe "bundle show" do
     end
 
     it "doesn't update gems to newer versions" do
-      install_gemfile <<-G
+      install_gemfile! <<-G
         source "file://#{gem_repo2}"
         gem "rails"
       G
 
+      expect(the_bundle).to include_gem("rails 2.3.2")
+
       update_repo2 do
         build_gem "rails", '3.0.0' do |s|
           s.executables = "rails"
         end
       end
 
-      bundle "show --outdated"
+      bundle! "show --outdated"
 
-      bundle "install"
-      expect(the_bundle).to include_gems("rails 2.3.2")
+      bundle! "install"
+      expect(the_bundle).to include_gem("rails 2.3.2")
     end
   end
 end
```

**Comment by colby-swandale** [comment]
@segiddins nice 👍

File: spec/commands/show_spec.rb [hunk 2]
```diff
@@ -169,21 +169,23 @@
...
-      install_gemfile <<-G
+      install_gemfile! <<-G
         source "file://#{gem_repo2}"
         gem "rails"
       G
 
+      expect(the_bundle).to include_gem("rails 2.3.2")
+
       update_repo2 do
         build_gem "rails", '3.0.0' do |s|
```
Comment (segiddins) [inline spec/commands/show_spec.rb]: double quotes

Comment (colby-swandale) [inline spec/commands/show_spec.rb]: my bad

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**